x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/home/index.tsx

```diff
@@ -69,7 +89,26 @@ const InstalledPackages: React.FC = memo(() => {
   const { data: allPackages, isLoading: isLoadingPackages } = useGetPackages({
     experimental: true,
   });
-  const [selectedCategory, setSelectedCategory] = useState('');
+
+  const { selectedCategory, searchParam } = getParams(
+    useParams<CategoryParams>(),
+    useLocation().search
+  );
+  const history = useHistory();
+  function setSelectedCategory(categoryId: string) {
+    const url = pagePathGetters.integrations_installed({
+      category: categoryId,
+      query: searchParam,
+    })[1];
+    history.push(url);
+  }
+  function setSearchTerm(search: string) {
+    const url = pagePathGetters.integrations_installed({
+      category: selectedCategory,
+      query: search,
+    })[1];
+    history.push(url);
```

Comment:
I think we can solve this problem of creating a history entry on each key press by changing this to a `replace` so that the last search term will be preserved in the history, but we don't create a new entry for each keypress:

```suggestion
    history.replace(url);
```

Reply:
TIL history.replace 🙇‍♂️